mkmapgrids/mkscripgrid.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri May 31 11:44:57 2024
+
+@author: olgad
+"""
+
+# This script creates a SCRIP grid file for a land-only point or regional case.
+# It is a python version of mkscripgrid.ncl originally written by Erik Kluzek Dec/07/2011
+
+import os
+import numpy as np
+from datetime import datetime
+from netCDF4 import Dataset
+
+def getenv_double(var_name, default):
+    value = os.getenv(var_name)
+    return float(value) if value else default
+
+def getenv_int(var_name, default):
+    value = os.getenv(var_name)
+    return int(value) if value else default
+
+def getenv_str(var_name, default):
+    value = os.getenv(var_name)
+    return value if value else default
+
+def fspan1up(fbegin, fend, number):
+    if number == 1:
+        return np.array([(fbegin + fend) / 2.0])
+    else:
+        return np.linspace(fbegin, fend, number)
+
+# ===========================================================================================================
+# Set a few constants needed later
+cdate = datetime.now().strftime("%y%m%d")
+ldate = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+# IMPORTANT NOTE: EDIT THE FOLLOWING TO CUSTOMIZE.
+
+# Input resolution and position
+name = getenv_str("PTNAME", None) # Set your grid name here
+
+latS = getenv_double("S_LAT", 45.5098) # Set the south latitude here
+latN = getenv_double("N_LAT", 45.6098) # Set the north latitude here
+lonE = getenv_double("E_LON", 275.3362) # Set the east longitude here
+lonW = getenv_double("W_LON", 275.2362) # Set the west longitude here
+
+nx = getenv_int("NX", 1) # Set the number of grids along longitude lines here
+ny = getenv_int("NY", 1) # Set the number of grids along latitude lines here
+
+
+imask = getenv_int("IMASK", 1)
+
+print_str = getenv_str("PRINT", "TRUE")
+printn = True if print_str == "TRUE" else False
+
+outfilename = getenv_str("GRIDFILE", None)
+#gitdescribe = getenv_str("GITDES", os.popen("git describe").read().strip())
+
+if name is None:
+    name = f"{nx}x{ny}pt_US-UMB"
+
+if outfilename is None:
+    if imask == 1:
+        outfilename = f"SCRIPgrid_{name}_nomask_c{cdate}.nc"
+    elif imask == 0:
+        outfilename = f"SCRIPgrid_{name}_noocean_c{cdate}.nc"
+    else:
+        outfilename = f"SCRIPgrid_{name}_mask_c{cdate}.nc"
+
+os.system(f"/bin/rm -f {outfilename}")
+if printn:
+    print(f"output file: {outfilename}")
+
+# Compute derived quantities
+delX = (lonE - lonW) / nx
```

Comment:
There might be problems when passing the prime meridian with this line, e.g. (10-350)/nx != 20/nx

Reply:
We could just add a if statement, when to add 360 when `lonE` or `lonW` is set below 0 and the program will run through smoothly.